libpromises/evalfunction.c

```diff
@@ -2733,10 +2756,29 @@ static FnCallResult FnCallDatastate(EvalContext *ctx,
                                     ARG_UNUSED const FnCall *fp,
                                     ARG_UNUSED const Rlist *args)
 {
-    JsonElement *state = DefaultTemplateData(ctx);
+    JsonElement *state = DefaultTemplateData(ctx, NULL);
     return  (FnCallResult) { FNCALL_SUCCESS, (Rval) { state, RVAL_TYPE_CONTAINER } };
 }
 
+static FnCallResult FnCallBundlestate(EvalContext *ctx,
+                                      ARG_UNUSED const Policy *policy,
+                                      ARG_UNUSED const FnCall *fp,
+                                      ARG_UNUSED const Rlist *args)
+{
+    JsonElement *state = DefaultTemplateData(ctx, RlistScalarValue(args));
+
+    if (NULL == state ||
+        JsonGetElementType(state) != JSON_ELEMENT_TYPE_CONTAINER ||
+        JsonLength(state) < 1)
+    {
+        return FnFailure();
```

Comment:
Ah yes, in case convergence will make it succeed later... I'm not entirely sure either, to be honest, but I assume your proposed behavior is the safest in that case.